python/pyspark/sql/dataframe.py

```diff
@@ -1272,16 +1272,18 @@ def replace(self, to_replace, value, subset=None):
         """Returns a new :class:`DataFrame` replacing a value with another value.
         :func:`DataFrame.replace` and :func:`DataFrameNaFunctions.replace` are
         aliases of each other.
+        Values `to_replace` and `value` should be homogeneous. Mixed string and numeric
```

Comment:
Do you have a reproducible example of this behavior? Scala seems to accept mixed numerics (though it is not documented):
```scala
val df = Seq(("Alice", 1, 4.1)).toDF
df.na.replace(df.columns, Map[Any, Any](1 -> 2, 4.1 -> 5.1)).
```
and Python equivalent seems to work for me as well.